src/components/AccountMenu.tsx

```diff
@@ -0,0 +1,137 @@
+'use client'
+import * as React from 'react'
+import Box from '@mui/material/Box'
+import Avatar from '@mui/material/Avatar'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import Divider from '@mui/material/Divider'
+import IconButton from '@mui/material/IconButton'
+import Tooltip from '@mui/material/Tooltip'
+import CurrencyExchangeIcon from '@mui/icons-material/CurrencyExchangeTwoTone'
+import LogoutIcon from '@mui/icons-material/LogoutTwoTone'
+import FavoriteIcon from '@mui/icons-material/FavoriteTwoTone'
+import ShoppingBasketIcon from '@mui/icons-material/ShoppingBasketTwoTone'
+import ContrastIcon from '@mui/icons-material/ContrastTwoTone'
+import { Switch } from '@mui/material'
+import Link from 'next/link'
+import TransitionsModal from './TransitionModal'
+import SettingsIcon from '@mui/icons-material/SettingsTwoTone'
+import ExchangeModal from './ExchangeModal'
+
+export default function AccountMenu() {
+	const [modalOpen, setModalOpen] = React.useState(false)
+	const [contentModal, setContentModal] = React.useState<React.ReactNode>(null)
+
+	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+	const open = Boolean(anchorEl)
```

Comment:
I don't really get why we need to use the state for the anchorEl. We can just use the event.currentTarget directly in the handleClick function.